rules-reviewed/technology-usage/security-technology-usage.windup.xml

```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0"?>
+<ruleset id="technology-usage-security" xmlns="http://windup.jboss.org/schema/jboss-ruleset" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides statistical summaries of the security related items in the input application (eg, security realms).
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <targetTechnology id="eap"/>
+        <phase>PostMigrationRulesPhase</phase>
+    </metadata>
+    <rules>
+        <rule id="technology-usage-security-01000">
+            <when>
+                <or>
+                    <xmlfile matches="//*[local-name() = 'login-config']/*[local-name() = 'realm-name']"
+                            in="web.xml"/>
+
+                    <xmlfile matches="//*[local-name() = 'login-config']/*[local-name() = 'realm']"
+                            in="{*}ejb-jar.xml"/>
+
+                    <javaclass references="javax.annotation.security.DeclareRoles">
+                        <location>ANNOTATION</location>
+                    </javaclass>
+
+                    <javaclass references="javax.annotation.security.RolesAllowed">
```

Comment:
I wonder if, from the "Security Realm" perspective, any annotation from `javax.annotation.security` should count as an instance of that technology and not only these two.

Reply:
the question is how is the request for tech statistics stated. Is it to count with Realms defined/used or count with usage occurences doesn't matter if 1 or more realms.

Reply:
I do not think a count is particularly important. I think it can scan all of the annotations.

Reply:
Fixed :)